### PR TITLE
Fix the flaky test_model_multiple_pcollections_partition test

### PR DESCRIPTION
**Pull request by aaltay**
_NativeWriteEvaluator should ignore empty bundles that arrive after a
write. Write happens once the last bundle containing data is processed.

### CODE REVIEW

**Comment by aaltay** [comment]
Locally run the test 100 times with no errors.

R: @charlesccychen

**Comment by aaltay** [comment]
@charlesccychen ping?

**Comment by charlesccychen** [comment]
Thanks, LGTM.

**Comment by aaltay** [comment]
Thank you!

@kennknowles could you please merge?

File: sdks/python/apache_beam/runners/direct/transform_evaluator.py [hunk 1]
```diff
@@ -524,15 +529,19 @@ def process_element(self, element):
...
-      if isinstance(self._sink, io.fileio.NativeTextFileSink):
-        assert self._sink.num_shards in (0, 1)
-        if self._sink.shard_name_template:
-          self._sink.file_path += '-00000-of-00001'
-          self._sink.file_path += self._sink.file_name_suffix
-      self._sink.pipeline_options = self._evaluation_context.pipeline_options
-      with self._sink.writer() as writer:
-        for v in self.state:
-          writer.Write(v.value)
+      if not self._has_already_produced_output:
```
Comment (kennknowles) [inline sdks/python/apache_beam/runners/direct/transform_evaluator.py]: I understand this is a surgical fix, but it is worth a comment that three nested conditionals is getting a bit hard to read, especially with the distance between them. I'd at least drop the `not` and put the short branch first so the `if` and `else` are more adjacent.

Comment (aaltay) [inline sdks/python/apache_beam/runners/direct/transform_evaluator.py]: Done.

**Comment by aaltay** [comment]
@kennknowles , thank your the review and sorry for the lack of context. 

The test in the PR title is flaky. The reason is one of the transforms result in at least one empty bundle. And depending on the thread scheduling, sometimes this empty bundle arrives as the last bundle to the native write transform.

DirectPipelineRunner implements native write transform by accumulating all data carrying bundles and then writing them as a single shard. Empty bundles arriving after the write happened was not handled before and it was resulting in overwriting the output files with empty data.

Please take another look.

**Comment by kennknowles** [comment]
OK, LGTM. I'll merge it.

**Pull request merged**